test/common/tracing/zipkin/zipkin_tracer_impl_test.cc

```diff
@@ -243,33 +243,114 @@ TEST_F(ZipkinDriverTest, FlushSpansTimer) {
   EXPECT_EQ(1U, stats_.counter("tracing.zipkin.spans_sent").value());
 }
 
-TEST_F(ZipkinDriverTest, SerializeAndDeserializeContext) {
+TEST_F(ZipkinDriverTest, NoB3ContextSampledTrue) {
   setupValidDriver();
 
-  // Supply bogus context, that will be simply ignored.
-  const std::string invalid_context = "notvalidcontext";
-  request_headers_.insertOtSpanContext().value(invalid_context);
-  driver_->startSpan(config_, request_headers_, operation_name_, start_time_);
+  EXPECT_EQ(nullptr, request_headers_.XB3SpanId());
+  EXPECT_EQ(nullptr, request_headers_.XB3TraceId());
+  EXPECT_EQ(nullptr, request_headers_.XB3Sampled());
 
-  std::string injected_ctx = request_headers_.OtSpanContext()->value().c_str();
-  EXPECT_FALSE(injected_ctx.empty());
+  Tracing::SpanPtr span = driver_->startSpan(config_, request_headers_, operation_name_,
+                                             start_time_, {Tracing::Reason::Sampling, true});
 
-  // Supply empty context.
-  request_headers_.removeOtSpanContext();
-  Tracing::SpanPtr span =
-      driver_->startSpan(config_, request_headers_, operation_name_, start_time_);
+  ZipkinSpanPtr zipkin_span(dynamic_cast<ZipkinSpan*>(span.release()));
+  EXPECT_TRUE(zipkin_span->span().sampled());
+}
+
+TEST_F(ZipkinDriverTest, NoB3ContextSampledFalse) {
```

Comment:
should a test class include one for absence of XB3Sampled from inbound headers? (I don't see it but might be blind)

Reply:
The following test does - the trace and span id are set, but the sampling decision is taken from envoy.